common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/type/TypeCreator.java

```diff
@@ -45,28 +40,34 @@ public TypeCreator(ReferenceCreator referenceCreator, FieldCreator fieldCreator,
     @Override
     protected void addFields(Type type, ClassInfo classInfo, Reference reference) {
         // Fields
-        List<MethodInfo> allMethods = new ArrayList<>();
+        List<List<MethodInfo>> allMethods = new LinkedList<>();
         Map<String, FieldInfo> allFields = new HashMap<>();
 
         // Find all methods and properties up the tree
         for (ClassInfo c = classInfo; c != null; c = ScanningContext.getIndex().getClassByName(c.superName())) {
+            List<MethodInfo> listMethods = new ArrayList<>();
+
             if (InterfaceCreator.canAddInterfaceIntoScheme(c.toString())) { // Not java objects
                 List<MethodInfo> classMethods = filterOutBridgeMethod(c.methods());
-                allMethods.addAll(classMethods);
-                allMethods.addAll(getAllInterfaceMethods(c, classMethods
+                listMethods.addAll(getAllInterfaceMethods(c, classMethods
                         .stream()
                         .map(MethodInfo::toString)
                         .collect(Collectors.toSet())));
+                listMethods.addAll(classMethods);
                 for (FieldInfo fieldInfo : c.fields()) {
                     allFields.putIfAbsent(fieldInfo.name(), fieldInfo);
                 }
             }
+
+            allMethods.add(0, listMethods);
```

Comment:
Always `0`? I assume you want some `i` that is incremented in each execution of the enclosing `for` loop

Reply:
Or just simply call `add(listMethods)` without specifying an index

Reply:
Ah, my bad, I thought this was always overwriting the element on that index, I had to re-read the javadoc.